crates/router/src/db/payment_method.rs

```diff
@@ -61,6 +72,297 @@ pub trait PaymentMethodInterface {
+
+                let key = format!("mid_{}_cust_{}", merchant_id, customer_id);
+                let field = format!("payment_method_id_{}", payment_method_new.payment_method_id);
+
+
+                let reverse_lookup_entry = |v: String| diesel_models::ReverseLookupNew {
+                    sk_id: field.clone(),
+                    pk_id: key.clone(),
+                    lookup_id: v,
+                    source: "payment_method".to_string(),
+                    updated_by: storage_scheme.to_string(),
+                };
+
+                let lookup_id1 = format!("payment_method_{}", &payment_method_new.payment_method_id);
+                let mut reverse_lookups = vec![lookup_id1];
+                if let Some(locker_id) = &payment_method_new.locker_id{
+                    reverse_lookups.push(format!("payment_method_locker_{}", locker_id))
+                }
+
+                let results = reverse_lookups
+                    .into_iter()
+                    .map(|v| 
+                        self.insert_reverse_lookup(reverse_lookup_entry(v), storage_scheme)
+                    );
+                
+                futures::future::try_join_all(results).await?;
+
+                let storage_payment_method = diesel_models::PaymentMethod::from(&payment_method_new);
+
+                let redis_entry = kv::TypedSql {
+                    op: kv::DBOperation::Insert {
+                        insertable : kv::Insertable::PaymentMethod(payment_method_new),
+                    },
+                };
+
+                match kv_wrapper::<diesel_models::PaymentMethod, _, _>(
+                    self,
+                    KvOperation::<diesel_models::PaymentMethod>::HSetNx(
+                        &field,
+                        &storage_payment_method,
+                        redis_entry,
+                    ),
+                    &key,
+                )
+                .await
+                .map_err(|err| err.to_redis_failed_response(&key))?
+                .try_into_hsetnx()
+                {
+                    Ok(HsetnxReply::KeyNotSet) => Err(errors::StorageError::DuplicateValue {
+                        entity: "payment_method",
+                        key: Some(storage_payment_method.payment_method_id),
+                    }
+                    .into()),
+                    Ok(HsetnxReply::KeySet) => Ok(storage_payment_method),
+                    Err(er) => Err(er).change_context(errors::StorageError::KVError),
+                }
+
+            }
+        }
+        
+    }
+
+    #[instrument(skip_all)]
+    async fn update_payment_method(
+        &self,
+        payment_method: storage::PaymentMethod,
+        payment_method_update: storage::PaymentMethodUpdate,
+        storage_scheme: MerchantStorageScheme,
+    ) -> CustomResult<storage::PaymentMethod, errors::StorageError> {
+        let conn = connection::pg_connection_write(self).await?;
```

Comment:
This connection is redundant when we are using KV we can move it undr PostgresOnly

Reply:
sure